src/test/java/com/microsoft/sqlserver/jdbc/NTLMConnectionTest.java

```diff
@@ -0,0 +1,689 @@
+        String password = System.getProperty("passwordNTLM");
+
+        if (null != domain) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "domain", domain);
+            ntlmPropsDefined = true;
+        }
+
+        if (null != user) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "user", user);
+            ntlmPropsDefined = true;
+        }
+
+        if (null != password) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "password", password);
+            ntlmPropsDefined = true;
+        }
+
+        if (ntlmPropsDefined) {
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "authenticationScheme", "NTLM");
+            connectionStringNTLM = addOrOverrideProperty(connectionStringNTLM, "integratedSecurity", "true");
+
+            dsNTLMLocal = new SQLServerDataSource();
+            updateDataSource(connectionStringNTLM, dsNTLMLocal);
+        }
+    }
+
+    /**
+     * Tests basic NTLM authentication.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMBasicConnection() throws SQLException {
+        if (ntlmPropsDefined) {
+            try (Connection con1 = dsNTLMLocal.getConnection();
+                    Connection con2 = DriverManager.getConnection(connectionStringNTLM)) {
+                verifyNTLM(con1);
+                verifyNTLM(con2);
+            }
+        }
+    }
+
+    /**
+     * Tests invalid connection property combinations.
+     */
+    @Test
+    public void testNTLMInvalidProperties() {
+        if (ntlmPropsDefined) {
+            // NTLM without user name
+            testInvalidProperties(addOrOverrideProperty(connectionStringNTLM, "user", ""),
+                    "R_NtlmNoUserPasswordDomain");
+
+            // NTLM without password
+            testInvalidProperties(addOrOverrideProperty(connectionStringNTLM, "password", ""),
+                    "R_NtlmNoUserPasswordDomain");
+
+            // NTLM with integratedSecurity property
+            testInvalidProperties(
+                    addOrOverrideProperty(connectionStringNTLM, "authentication", "ActiveDirectoryIntegrated "),
+                    "R_SetAuthenticationWhenIntegratedSecurityTrue");
+        }
+    }
+
+    /**
+     * Tests NTLM authentication when the connection is encrypted.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMEncryptedConnection() throws SQLException {
+        if (ntlmPropsDefined) {
+            try (Connection con = DriverManager
+                    .getConnection(connectionStringNTLM + ";encrypt=true;trustServerCertificate=true")) {
+                verifyNTLM(con);
+            }
+        }
+    }
+
+    /**
+     * Tests NTLM authentication against a non-default database.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMNonDeafultDatabase() throws SQLException {

```

Comment:
I changed the test a bit to not use a system database and NTLM connection seems to fail.

```
    @Test
    public void testNTLMNonDeafultDatabase() throws SQLException {
        if (ntlmPropsDefined) {
            String databaseName = RandomUtil.getIdentifier("NTLM");
            try (Statement stmt = connection.createStatement()) {
                stmt.executeUpdate("CREATE DATABASE " + AbstractSQLGenerator.escapeIdentifier(databaseName));
            }
            try (Connection con = DriverManager
                    .getConnection(addOrOverrideProperty(connectionStringNTLM, "database", databaseName))) {
                verifyNTLM(con);
                verifyDatabase(con, databaseName);
            }
            try (Statement stmt = connection.createStatement()) {
                TestUtils.dropDatabaseIfExists(databaseName, stmt);
            }
        }
    }
```

Reply:
The test is fine and was failing because of a permission issue on my server. Please update `testNTLMNonDeafultDatabase()` test to not use `tempdb`.